src/Build.UnitTests/BackEnd/TaskHostFactory_Tests.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using Microsoft.Build.Execution;
+using Microsoft.Build.UnitTests;
+using Shouldly;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Build.Engine.UnitTests.BackEnd
+{
+    public sealed class TaskHostFactory_Tests
+    {
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp, "https://github.com/microsoft/msbuild/issues/5158")]
+        public void TaskNodesDieAfterBuild()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                string pidTaskProject = $@"
+<Project>
+    <UsingTask TaskName=""ProcessIdTask"" AssemblyName=""Microsoft.Build.Engine.UnitTests"" TaskFactory=""TaskHostFactory"" />
+    <Target Name='AccessPID'>
+        <ProcessIdTask>
+            <Output PropertyName=""PID"" TaskParameter=""pid"" />
+        </ProcessIdTask>
+    </Target>
+</Project>";
+                TransientTestFile project = env.CreateFile("testProject.csproj", pidTaskProject);
+                ProjectInstance projectInstance = new ProjectInstance(project.Path);
+                projectInstance.Build().ShouldBeTrue();
+                Should.Throw<ArgumentException>(() => Process.GetProcessById(Int32.Parse(projectInstance.GetPropertyValue("PID"))));
```

Comment:
I'd split this up to help with diagnosability when it fails (as it's doing now).

```suggestion
                string pidProperty = projectInstance.GetPropertyValue("PID");
                string.IsNullOrEmpty(pidProperty).ShouldBeFalse();

                // also log the PID that we're looking for

                int.TryParse(pidProperty, out int pid).ShouldBeTrue();

                Should.Throw<ArgumentException>(() => Process.GetProcessById(pid));
```